soc/arm/nxp_imx/mimx8mm6_m4/soc.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2020, Manivannan Sadhasivam <mani@kernel.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <fsl_clock.h>
+#include <fsl_common.h>
+#include <fsl_rdc.h>
+#include <init.h>
+#include <kernel.h>
+#include <soc.h>
+
+#include <dt-bindings/rdc/imx_rdc.h>
+
+/* OSC/PLL is already initialized by ROM and Cortex-A7 (u-boot) */
+static void SOC_RdcInit(void)
+{
+	/* Move M4 core to specific RDC domain 1 */
+	rdc_domain_assignment_t assignment = {0};
+
+	assignment.domainId = M4_DOMAIN_ID;
+	RDC_SetMasterDomainAssignment(RDC, kRDC_Master_M4, &assignment);
+
+	/*
+	 * The M4 core is running at domain 1, enable clock gate for
+	 * Iomux to run at domain 1.
+	 */
+	CLOCK_EnableClock(kCLOCK_Iomux0);
+	CLOCK_EnableClock(kCLOCK_Iomux1);
+	CLOCK_EnableClock(kCLOCK_Iomux2);
+	CLOCK_EnableClock(kCLOCK_Iomux3);
+	CLOCK_EnableClock(kCLOCK_Iomux4);
+
+	CLOCK_EnableClock(kCLOCK_Qspi);
+
+	/*
+	 * The M4 core is running at domain 1, enable the PLL clock sources
+	 * to domain 1.
+	 */
+	/* Enable SysPLL1 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_SysPll1Gate, kCLOCK_ClockNeededAll);
+	/* Enable SysPLL2 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_SysPll2Gate, kCLOCK_ClockNeededAll);
+	/* Enable SysPLL3 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_SysPll3Gate, kCLOCK_ClockNeededAll);
+	/* Enable AudioPLL1 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_AudioPll1Gate, kCLOCK_ClockNeededAll);
+	/* Enable AudioPLL2 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_AudioPll2Gate, kCLOCK_ClockNeededAll);
+	/* Enable VideoPLL1 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_VideoPll1Gate, kCLOCK_ClockNeededAll);
+}
+
+/* AUDIO PLL1 configuration */
+static const ccm_analog_frac_pll_config_t g_audioPll1Config = {
```

Comment:
Are all of these PLLs configurations applicable to all boards that use this SoC ? Seems that the divisors might change if the same SoC is used with a different board, which probably makes this board-specific stuff instead of SoC-specific.. Can you confirm ?